easypy/signals.py

```diff
@@ -312,16 +354,32 @@ def register_object(obj, **kwargs):
         params.update(kwargs)
 
         method_name, *_ = method_name.partition("__")  # allows multiple methods for the same signal
+
+        if not inspect.ismethod(method):
+            # Fake method attributes for WeakMethod
+            method.__self__ = obj
+            method.__func__ = getattr(type(obj), method_name)
+            if not hasattr(method, '__code__'):
+                method.__code__ = __FakeCode('', '')
+            fake_meth_type = functools.partial(
+                __fake_meth_type,
+                # Best effort: fake name and code for SignalHandler. We don't know the line number.
+                name=method_name,
+                code=__FakeCode(type(obj).__module__, '?'))
+
+        method = weakref.WeakMethod(method)
+        try:
+            method._meth_type = fake_meth_type
+        except UnboundLocalError:
+            pass
+
         register_signal(method_name, method, **params)
 
 
 def unregister_object(obj):
-    for method_name in get_signals_for_type(type(obj)):
-        method = getattr(obj, method_name)
-        if not callable(method):
-            continue
-        method_name, *_ = method_name.partition("__")
-        unregister_signal(method_name, method)
+    for signal_name in {method_name.split('__', 1)[0] for method_name in get_signals_for_type(type(obj))}:
+        for handlers in Signal.ALL[signal_name].handlers.values():
+            handlers[:] = (handler for handler in handlers if handler.bound_object is not obj)
```

Comment:
I've changed this because I think the current object unregistration may be too slow - see https://wekaio.atlassian.net/browse/WEKAPP-73925